services/movie.py

```diff
@@ -1,37 +1,44 @@
+from django.db import transaction
 from django.db.models import QuerySet
+from django.shortcuts import get_object_or_404
 
+import init_django_orm  # noqa: F401
 from db.models import Movie
 
 
 def get_movies(
-    genres_ids: list[int] = None,
-    actors_ids: list[int] = None,
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None,
+        title: str = None
 ) -> QuerySet:
     queryset = Movie.objects.all()
 
     if genres_ids:
         queryset = queryset.filter(genres__id__in=genres_ids)
-
     if actors_ids:
         queryset = queryset.filter(actors__id__in=actors_ids)
+    if title:
+        queryset = queryset.filter(title__icontains=title).values_list("title")
 
     return queryset
 
 
 def get_movie_by_id(movie_id: int) -> Movie:
-    return Movie.objects.get(id=movie_id)
+    return get_object_or_404(Movie, id=movie_id)
 
 
+@transaction.atomic
 def create_movie(
-    movie_title: str,
-    movie_description: str,
-    genres_ids: list = None,
-    actors_ids: list = None,
+        movie_title: str,
+        movie_description: str,
+        genres_ids: list = None,
+        actors_ids: list = None,
```

Comment:
Consider using `Optional[list[int]]` from the `typing` module for `genres_ids` and `actors_ids` to explicitly indicate that these parameters can be `None`.

Reply:
This has already been written by the mentor